src/test/components/AppLocalizationProvider.test.js

```diff
@@ -43,36 +38,19 @@ describe('AppLocalizationProvider', () => {
 
     const translatedText = (language) => `This is ${language} Text`;
     const fetchUrlRe = /^\/locales\/(?<language>[^/]+)\/app.ftl$/;
-    const fetch = jest.fn().mockImplementation((fetchUrl: string) => {
-      const matchUrlResult = fetchUrlRe.exec(fetchUrl);
-      if (matchUrlResult) {
-        // $FlowExpectError Our Flow doesn't know about named groups.
-        const { language } = matchUrlResult.groups;
-        const response = (({
-          ok: true,
-          status: 200,
-          headers: {
-            get: () => 'text/plain',
-          },
-          text: () =>
-            Promise.resolve(
-              missingTranslation.includes(language)
-                ? ``
-                : `test-id = ${translatedText(language)}`
-            ),
-        }: any): Response);
-
-        return Promise.resolve(response);
-      }
-      return Promise.reject(
-        coerceMatchingShape<Response>({
-          ok: false,
-          status: 404,
-          statusText: 'Not found',
-        })
-      );
-    });
-    (window: any).fetch = fetch;
+    window.fetch
+      .catch(404) // catchall
```

Comment:
The default for `catch` is to return an empty 200 response. That's why I specify it everywhere,

However whenever a request goes to the `catch` there's a warning to the console (this can be disabled). So I tried to specify the rules for every tests so that `catch` isn't even reached.

While changing this I found the warning useful, that's why I decided to keep it. But we could possibly configure a global `catch` that would throw instead of returning 200.